src/snapred/backend/service/ReductionService.py

```diff
@@ -137,6 +172,9 @@ def reduction(self, request: ReductionRequest):
         ingredients = self.prepReductionIngredients(request)
 
         groceries = self.fetchReductionGroceries(request)
+        if isinstance(groceries, ArtificialNormResponse):
+            return groceries
```

Comment:
Again we shouldnt be expanding type like this.  I'll probably have another comment when I get to `prepReductionIngredients`.